File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/ProfileStreamEvent.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.playerWithCard.PlayerWithCardsModel;

public interface ProfileStreamEvent {
    void onConnected(ProfileStreamEventController c);
    void onGetPlayerData(PlayerWithCardsModel p);
    void onPlayerCardUpdated();
    void onDisconnected();
    void onError(String e);
    void onException( String e,int flag,ProfileStreamEventController c);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/ProfileStreamEvent.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.playerWithCard.PlayerWithCardsModel

interface ProfileStreamEvent {
    fun onConnected(c : ProfileStreamEventController)
    fun onGetPlayerData(p : PlayerWithCardsModel)
    fun onPlayerCardUpdated()
    fun onDisconnected()
    fun onError(e : String)
    fun onException(e : String,flag : Int,c : ProfileStreamEventController)
}